File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] arr) {
        int l=0, r=arr.length-1;
        int result = 0;
        while(l<=r) {
            int area = (r-l)*Math.min(arr[l], arr[r]);
            result = Math.max(area, result);
            if(arr[l]<arr[r]) l++;
            else r--;
        }
        return result;
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        int dp[n+1] = {0};
        for(int i=2; i<=n; i++) {
            dp[i] =  min(dp[i-1]+cost[i-1], dp[i-2]+cost[i-2]);
        }
        for(auto i : dp) cout<<i<<' ';
        cout<<endl;
        return dp[n];
    }
};
